metagraph/src/cli/align.cpp

```diff
@@ -433,10 +434,20 @@ int align_to_graph(Config *config) {
                 num_bytes_read += it->seq.l;
             }
 
-            auto process_batch = [&](SeqBatch batch) {
-                auto aln_graph = graph;
-                if (auto *canonical = dynamic_cast<CanonicalDBG*>(graph.get()))
-                    aln_graph = std::make_shared<CanonicalDBG>(*canonical);
+            auto process_batch = [&,base_graph](SeqBatch batch) {
+                // aliasing constructor
+                auto aln_graph = base_graph;
+                auto *dbg_succ = dynamic_cast<DBGSuccinct*>(aln_graph.get());
+                bool is_primary = aln_graph->get_mode() == DeBruijnGraph::PRIMARY;
```

Comment:
why don't you always wrap primary into a canonical?

Reply:
`DBGSuccinctCached` needs to wrap on top of `DBGSuccinct` first, then `CanonicalDBG` on top of that.

Reply:
ok, what is going on in this function?

You take the base graph, check if it's primary ... ???

Eventually, you want to create another view on the graph with its own cache, right?

Move this to a separate function.

Reply:
done. I've also replaced the outdated comment to explain things a bit better